debezium-api/src/main/java/io/debezium/spi/topic/TopicNamingStrategy.java

```diff
@@ -43,17 +43,17 @@
      *
      * @link https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/internals/Topic.java
      */
-    default String sanitizedTopicName(String topicName) {
+    default String sanitizedTopicName(String topicName, ReplacementFunction replacement) {
```

Comment:
I don't think we should change the signature and add the replacement function here.

IMHO it is about time to move `sanitizedTopicName` implementation into `AbstractTopicNamingStrategy` and pass replacement function via factory method or constructor.